employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/skill/SkillService.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.service.skill;
+
+import java.util.List;
+
+import org.optaweb.employeerostering.domain.Skill;
+import org.optaweb.employeerostering.persistence.SkillRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SkillService {
+
+    @Autowired
+    private SkillRepository skillRepository;
+
+    public List<Skill> getSkillList(Integer tenantId) {
+        return skillRepository.findAll();
+    }
+
+    public Skill getSkill(Integer tenantId, Long id) {
+        if (skillRepository.findById(id).isPresent())
+            return skillRepository.findById(id).get();
+
+        return new Skill();
+    }
+
+    public void deleteSkill(Integer tenantId, Long id) {
+        if (skillRepository.findById(id).isPresent())
+            skillRepository.deleteById(id);
+    }
+
+    public void createSkill(Integer tenantId, Skill skill) {
+        skillRepository.save(skill);
+    }
+
+    public void updateSkill(Integer tenantId, Skill skill) {
+        if (skill.getId() != null) {
+            if (skillRepository.findById(skill.getId()).isPresent()) {
+                Skill skill1 = skillRepository.findById(skill.getId()).get();
+
+                skill1.setTenantId(skill.getTenantId());
```

Comment:
Is this a legal operation? Does the domain allow any entity to move from one tenant to another? Please discuss this with @Christopher-Chianelli. The answer strongly depends on the definition of tenant. If the domain treats tenants as completely separated worlds, it's likely not possible to move entities between tenants. In that case making a call to the service with `tenantId` different from `skill.tenantId` would be an invalid request. In such case, write a defensive code and fail immediately. And don't forget about tests :slightly_smiling_face:

Reply:
I did not consider this, thank you for the reminder. I will discuss this with Chris.